### PR TITLE
Add and document makeRelease.sh script and composer command

### PR DESCRIPTION
**Pull request by arothuis**
Added simple script, based on existing scripts within the platform, to create deployment tarballs.
Added documentation.

### CODE REVIEW

**Comment by thijskh** [comment]
This looks good to me!

File: makeRelease.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+echo "Tagging the release in RELEASE file" &&
+COMMITHASH=`git rev-parse HEAD` &&
+echo "Tag: ${TAG}" > ${PROJECT_DIR}/RELEASE &&
+echo "Commit: ${COMMITHASH}" >> ${PROJECT_DIR}/RELEASE &&
+
+echo "Cleaning build of dev files" &&
+rm -rf ${PROJECT_DIR}/ansible &&
+rm -rf ${PROJECT_DIR}/Vagrantfile &&
+rm -rf ${PROJECT_DIR}/.idea &&
```
Comment (DRvanR) [inline makeRelease.sh]: is this in the project when doing a checkout? Should be `.gitignore`-ed :wink:

File: makeRelease.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+rm -rf ${PROJECT_DIR}/.git &&
+rm -f ${PROJECT_DIR}/.gitignore &&
+rm -f ${PROJECT_DIR}/makeRelease.sh &&
+rm -f ${PROJECT_DIR}/bin/composer.phar &&
```
Comment (DRvanR) [inline makeRelease.sh]: don't think we have this, or the next 2 files? (`features` and `behat.yml`)

File: makeRelease.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+rm -rf ${PROJECT_DIR}/.travis.yml &&
+rm -rf ${PROJECT_DIR}/.travis.php.ini &&
+rm -f ${PROJECT_DIR}/web/app_dev.php &&
+
+echo "Removing application cache, logs, bootstrap and parameters" &&
+rm -f ${PROJECT_DIR}/app/bootstrap.php.cache &&
+rm -rf ${PROJECT_DIR}/app/cache/* &&
+rm -rf ${PROJECT_DIR}/app/logs/* &&
+rm -f ${PROJECT_DIR}/app/config/global_view_parameters.yml &&
+rm -f ${PROJECT_DIR}/app/config/parameters.yml &&
```
Comment (DRvanR) [inline makeRelease.sh]: these two parameter files only have `.dist` variants right?

**Comment by DRvanR** [comment]
a few nits, other than that :+1:

File: makeRelease.sh (hunk 1 above)
Comment (arothuis) [inline makeRelease.sh]: I copied this from another `makeRelease`. I know there are people that don't have it in their global gitignore, so I left it in, but I agree: they should be gitignored.

File: makeRelease.sh (hunk 2 above)
Comment (arothuis) [inline makeRelease.sh]: Nope. Should've removed it.

File: makeRelease.sh (hunk 3 above)
Comment (arothuis) [inline makeRelease.sh]: Ah right, they are .gitignored so they should be removed.

**Pull request merged**